### PR TITLE
Fix/sadjad cli fixes

### PR DESCRIPTION
**Pull request by sksadjad**
- removed file option for ecosystem submit
- printing if `-s`
- re-enabling onboard for participant
- added persist flag to everywhere we're issuing a vp, limited console logs with show flag

### CODE REVIEW

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 1]
```diff
@@ -363,7 +372,7 @@ export class GXComplianceClient implements IAgentPlugin {
         verifiableCredentials: [args.complianceVC, args.selfDescriptionVC],
         challenge: args.challenge ? args.challenge : GXComplianceClient.getDateChallenge(),
         domain: did ?? extractSubjectDIDFromVCs([args.selfDescriptionVC]),
-        persist: true,
+        persist: args.persist ? args.persist : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:375]: Not sure why this is needed everywhere, simply pas in args.persist

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 2]
```diff
@@ -339,14 +349,13 @@ export class GXComplianceClient implements IAgentPlugin {
         verifiableCredentials: [verifiableCredentialResponse.verifiableCredential, args.complianceVC, args.selfDescriptionVC],
         challenge: args.challenge ? args.challenge : GXComplianceClient.getDateChallenge(),
         domain: participantDid,
-        persist: true,
+        persist: args.persist ? args.persist : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:352]: Not sure why this is needed everywhere, simply pas in args.persist

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 3]
```diff
@@ -324,12 +332,14 @@ export class GXComplianceClient implements IAgentPlugin {
...
         verifiableCredentials: [args.selfDescriptionVC, args.complianceVC],
         domain: convertDidWebToHost(participantDid),
+        persist: args.persist ? args.persist : false,
       },
       context
     )
     const verifiableCredentialResponse = (await this.acquireComplianceCredential(
       {
         verifiablePresentation: uniqueVP.verifiablePresentation,
+        show: args.show ? args.show : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:342]: Not sure why this is needed everywhere, simply pass in args.show

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 4]
```diff
@@ -324,12 +332,14 @@ export class GXComplianceClient implements IAgentPlugin {
         keyRef: args.keyRef,
         verifiableCredentials: [args.selfDescriptionVC, args.complianceVC],
         domain: convertDidWebToHost(participantDid),
+        persist: args.persist ? args.persist : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:335]: Not sure why this is needed everywhere, simply pass in args.show

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 5]
```diff
@@ -291,12 +295,16 @@ export class GXComplianceClient implements IAgentPlugin {
...
+    args: {
+      show?: boolean
+      verifiablePresentation: VerifiablePresentation
+    },
     context: GXRequiredContext
   ): Promise<VerifiableCredentialResponse> {
     const complianceCredential = await this.submitComplianceCredential(
       {
         selfDescriptionVP: args.verifiablePresentation,
+        show: args.show ? args.show : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:307]: Not sure why this is needed everywhere, simply pass in args.show

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 6]
```diff
@@ -194,7 +198,7 @@ export class GXComplianceClient implements IAgentPlugin {
         // purpose: args.purpose,
         verifiableCredentials: [complianceCredential, serviceOffering.verifiableCredential],
         domain: did,
-        persist: true,
+        persist: args.persist ? args.persist : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:201]: Not sure why this is needed everywhere, simply pass in args.persist

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 7]
```diff
@@ -144,19 +147,20 @@ export class GXComplianceClient implements IAgentPlugin {
...
         verifiableCredentials: [selfDescription.verifiableCredential],
         domain: signInfo.participantDomain,
+        persist: args.persist ? args.persist : false,
       },
       context
     )
     const verifiableCredentialResponse = (await this.acquireComplianceCredential(
       {
         verifiablePresentation: uniqueVP.verifiablePresentation,
+        show: args.show ? args.show : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:163]: Not sure why this is needed everywhere, simply pass in args.show

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 8]
```diff
@@ -144,19 +147,20 @@ export class GXComplianceClient implements IAgentPlugin {
...
       context
     )
-    console.log(selfDescription.hash)
     const uniqueVP = await this.credentialHandler.issueVerifiablePresentation(
       {
         challenge: GXComplianceClient.getDateChallenge(),
         keyRef: signInfo.keyRef,
         verifiableCredentials: [selfDescription.verifiableCredential],
         domain: signInfo.participantDomain,
+        persist: args.persist ? args.persist : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:156]: Not sure why this is needed everywhere, simply pass in args.persist

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 9]
```diff
@@ -111,13 +113,14 @@ export class GXComplianceClient implements IAgentPlugin {
         verifiableCredentials: [selfDescribedVC],
         challenge: GXComplianceClient.getDateChallenge(),
         domain: signInfo.participantDomain,
-        persist: true,
+        persist: args.persist ? args.persist : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:116]: Not sure why this is needed everywhere, simply pass in args.persist

File: packages/gx-agent/src/agent/GXComplianceClient.ts [hunk 10]
```diff
@@ -111,13 +113,14 @@ export class GXComplianceClient implements IAgentPlugin {
...
       },
       context
     )
     return this.acquireComplianceCredential(
       {
         verifiablePresentation: uniqueVP.verifiablePresentation,
+        show: args.show ? args.show : false,
```
Comment (nklomp) [inline packages/gx-agent/src/agent/GXComplianceClient.ts:123]: Not sure why this is needed everywhere, simply pass in args.show

File: packages/gx-agent-cli/lib/ecosystem.ts [hunk 11]
```diff
@@ -125,11 +113,15 @@ ecosystem
         return
       }
       const selfDescription = await agent.onboardParticipantOnEcosystem({
-        ecosystemUrl: normalizeEcosystemConfigurationObject(ecosystemConfig).url,
+        ecosystemUrl: ecosystemConfig.url,
         selfDescriptionVC,
         complianceVC,
+        persist: cmd.persist === true,
+        show: cmd.persist === true,
```
Comment (nklomp) [inline packages/gx-agent-cli/lib/ecosystem.ts:120]: cmd.persist used for show value

**Review by nklomp — CHANGES_REQUESTED** [review]
See remarks

File: packages/gx-agent/src/agent/GXComplianceClient.ts (hunk 2 above)
Comment (sksadjad) [reply packages/gx-agent/src/agent/GXComplianceClient.ts]: changed to just: `args.persist`

File: packages/gx-agent/src/agent/GXComplianceClient.ts (hunk 3 above)
Comment (sksadjad) [reply packages/gx-agent/src/agent/GXComplianceClient.ts:342]: changed to just: `args.persist`

File: packages/gx-agent/src/agent/GXComplianceClient.ts (hunk 4 above)
Comment (sksadjad) [reply packages/gx-agent/src/agent/GXComplianceClient.ts]: changed to just: `args.persist`

File: packages/gx-agent/src/agent/GXComplianceClient.ts (hunk 6 above)
Comment (sksadjad) [reply packages/gx-agent/src/agent/GXComplianceClient.ts]: changed to just: `args.persist`

File: packages/gx-agent/src/agent/GXComplianceClient.ts (hunk 7 above)
Comment (sksadjad) [reply packages/gx-agent/src/agent/GXComplianceClient.ts]: changed to just: `args.show`

File: packages/gx-agent/src/agent/GXComplianceClient.ts (hunk 8 above)
Comment (sksadjad) [reply packages/gx-agent/src/agent/GXComplianceClient.ts]: changed to just: `args.persist`

File: packages/gx-agent/src/agent/GXComplianceClient.ts (hunk 9 above)
Comment (sksadjad) [reply packages/gx-agent/src/agent/GXComplianceClient.ts]: changed to just: `args.persist`

File: packages/gx-agent/src/agent/GXComplianceClient.ts (hunk 10 above)
Comment (sksadjad) [reply packages/gx-agent/src/agent/GXComplianceClient.ts]: changed to just: `args.show`

**Review by nklomp — APPROVED** [review]

**Pull request merged**